rollup/internal/orm/batch.go

```diff
@@ -413,6 +413,16 @@ func (o *Batch) UpdateCommitTxHashAndRollupStatus(ctx context.Context, hash stri
 		db = dbTX[0]
 	}
 	db = db.WithContext(ctx)
+
+	var currentBatch Batch
+	if err := db.Where("hash", hash).First(&currentBatch).Error; err != nil {
+		return fmt.Errorf("Batch.UpdateCommitTxHashAndRollupStatus error when querying current status: %w, batch hash: %v", err, hash)
+	}
+
+	if types.RollupStatus(currentBatch.RollupStatus) == types.RollupFinalized {
```

Comment:
Here should consider also `RollupFinalizing` no?

Reply:
added in [63f02ad](https://github.com/scroll-tech/scroll/pull/1638/commits/63f02ad36738fd64c8e980c806d5cc3099615bf6).